### PR TITLE
[EPROD-424] remove user registration from emvc

### PR DESCRIPTION
**Pull request by itsyaasir**
# Issue ticket

Issue ticket link: <>

### Code conventions

- [ ] I have performed a self-review of my code
- [ ] Every new function is documented
- [ ] Object names are auto explicative

### Security

- [ ] The PR does not break APIs backward compatibility
- [ ] The PR does not break the stable storage backward compatibility

### Testing

- [ ] Every function is properly unit tested
- [ ] I have added integration tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

**Review by GraDKh — APPROVED** [review]

**Review by ufoscout — APPROVED** [review]

**Review by ufoscout — CHANGES_REQUESTED** [review]
Is the `register-evm-agent` updated with this changes?

**Comment by itsyaasir** [comment]
Should we change the naming of the crate ?

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        Ok(())
+    }
+
+    async fn is_address_reserved(&self) -> Result<bool> {
+        let address: H160 = self.wallet.address().into();
+        info!("checking if {address} is already reserved...");
+        let args = Encode!(&self.register_canister_id, &address)?;
+        let res = self
+            .agent
+            .query(&self.evmc_canister_id, METHOD_ADDRESS_RESERVED)
```
Comment (ufoscout) [inline src/register-evm-agent/src/register_evm_agent/src/reservation.rs:80]: can this be done using the evmc-client?

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs (hunk 1 above)
Comment (ufoscout) [reply src/register-evm-agent/src/register_evm_agent/src/reservation.rs]: done

**Review by ufoscout — APPROVED** [review]

**Pull request merged**